Add new widget types for directories and output files

jbeezley opened the pull request:
For file outputs, I created a text `<input>` below the hierarchy widget for the user to type in the new item name.

![screen shot 2016-04-14 at 2 04 25 pm](https://cloud.githubusercontent.com/assets/31890/14538453/d69bfb22-0249-11e6-928e-93f5d21ecefd.png)

There is minimal validation done when submitting the modal.

![screen shot 2016-04-14 at 1 57 22 pm](https://cloud.githubusercontent.com/assets/31890/14538380/844c93b8-0249-11e6-8649-72bb8bd6b2b9.png)

The main control widget now shows a path string rather than an object id.

![screen shot 2016-04-14 at 1 58 07 pm](https://cloud.githubusercontent.com/assets/31890/14538497/106b6842-024a-11e6-8e8e-bd2a1830e61f.png)

Finally, the submit button actually `POST`'s to the analysis endpoint.  I think it is submitting correctly, but I haven't been able to get the celery working running correctly locally yet.

zachmullen commented:
Minor nit, it looks like in your Select button, you put the text inside the `<i>` tag rather than as a sibling of it ;)

jbeezley commented:
Fixed.  Also, I change the button text to `Save`.

cdeepakroy commented:
@jbeezley To test, I git cloned your branch (directory-io), went to the front-end page at `http://localhost:8080/histomicstk#ColorDeconvolution`, and tried to set `Output Image of Stain 1`. But i could not see your new widget. See snapshot below:

![new_widget](https://cloud.githubusercontent.com/assets/1923534/14544300/55165d64-0266-11e6-8aed-c5e713fffe5f.png)

Do i have to do anything else to get to see the new widget?

jbeezley commented:
Did you rerun `npm install`?

cdeepakroy commented:
@jbeezley i could see the new widget after running npm install. If i hit submit, is it supposed to run the job, or that is not implemented yet?

jbeezley commented:
It should POST the analysis to the server.  When I tried, I got an internal server error, which I assumed was because my worker isn't setup correctly, but there could be something else going wrong.  I'll look into it more a bit later.

cdeepakroy commented:
I tried running it using the submit button but it dint seem to do anything --- no new job was initiated at `http://localhost:8080/#jobs/` and saw nothing on the girder or girder_worker command-line.

jbeezley commented:
I just fixed a bug that may have caused silent validation errors and added an error dialog to the form submission in case something goes wrong. You will need to`npm install` again (as always with client code updates).

cdeepakroy commented:
@jbeezley It now runs the CLI successfully. 

LGTM! Can be merged.
